### ISSUE TITLE
Uploading New File of Downloadable Break existing customer order.

### ISSUE THREAD

**Issue by yokowasis** [boxbilling/boxbilling#496]
When I upload a new version (different file name) of Downloadable Product. The existing customer when they click "Download File" they will find 404 not found.

This can be fixed by changing the the value of filename column on table `service_downloadable`. 

Any way to fix this ?

**Comment by BenNottelling** [comment]
This error is caused by BB not using the file update function when the product already has a file

**Comment by yokowasis** [comment]
To update downloadable files, you have to upload the new file using ftp
using the same file names as the old one.

**Comment by BenNottelling** [comment]
Yeah, that's how you would have to do it with the currently broken code.
There is a separate function created to update the downloadable file, however it isn't used in the default interface and instead uses the function for creating a new upload, thus breaking it.

I'll have to investigate what the best fix is, but I at least know what is causing the problem
